src/libs/actions/Session/index.js

```diff
@@ -455,17 +455,18 @@ function signInWithValidateCode(accountID, code, twoFactorAuthCode, preferredLoc
         },
     ];
 
-    API.write(
-        'SigninUserWithLink',
-        {
-            accountID,
-            validateCode,
-            twoFactorAuthCode,
-            preferredLocale,
-            deviceInfo: getDeviceInfoForLogin(),
-        },
-        {optimisticData, successData, failureData},
-    );
+    const params = {
+        accountID,
+        validateCode,
+        preferredLocale,
+        deviceInfo: getDeviceInfoForLogin(),
+    };
+
+    // Pass twoFactorAuthCode to server only if it has a valid value, otherwise php might convert it to "null" as a string.
+    if (twoFactorAuthCode) {
+        params.twoFactorAuthCode = twoFactorAuthCode;
+    }
```

Comment:
It looks like we're manually passing `null` for the `twoFactorAuthCode ` in several places (but that we often pass a `preferredLocale`). Maybe a cleaner solution would be to update this function so that `twoFactorAuthCode` is the last param and defaults to `''` if it's not passed. Then we don't even need this `if`

It's more changes but ultimately I think it's better than passing `null` in a bunch of places

Reply:
👌

Reply:
I'm on them atm